### PR TITLE
Product Gallery: fix PHP fatal error

### LINKED ISSUE #53761: [Fatal] PHP Fatal error:  Uncaught Error: Call to a member function get_title() on null in | wc-template-functions.php:1668
### Prerequisites

- [ ] I have carried out troubleshooting steps and I believe I have found a bug.
- [ ] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

During 9.5.0 release, we have witnessed a fatal when the `woocommerce-additional-variation-images` is installed

[link to code](https://github.com/woocommerce/woocommerce/blob/90faf2cacfaf4a5f9f0ba196a2afd8fe91696c40/plugins/woocommerce/includes/wc-template-functions.php#L1668)

we witnessed ~248 fatals across 18 sites after the deploy from 2pm UTC to 330pm UTC on Mon Dec 16th, 2024

```
PHP Fatal error:  Uncaught Error: Call to a member function get_title() on null in /../plugins/woocommerce/9.5.0/includes/wc-template-functions.php:1668
Stack trace:
#0 /../wp-content/plugins/woocommerce-additional-variation-images/includes/class-wc-additional-variation-images-frontend.php(101): wc_get_gallery_image_html(727, false)
#1 /../core/6.7.1/wp-includes/class-wp-hook.php(324): WC_Additional_Variation_Images_Frontend->ajax_get_images('')
#2 /../core/6.7.1/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters('', Array)
#3 /../core/6.7.1/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#4 /../plugins/woocommerce/9.5.0/includes/class-wc-ajax.php(96): do_action('wc_ajax_wc_addi...')
#5 /../core/6.7.1/wp-includes/class-wp-hook.php(324): WC_AJAX::do_wc_ajax('')
#6 /../core/6.7.1/wp-includes/class-wp-hook.php(348): WP_Hook->apply_filters(NULL, Array)
#7 /../core/6.7.1/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#8 /../core/6.7.1/wp-includes/template-loader.php(13): do_action('template_redire...')
#9 /../core/6.7.1/wp-blog-header.php(19): require_once('/../core...')
#10 /../core/6.7.1/index.php(17): require('/../core...')
#11 {main}
  thrown in /../plugins/woocommerce/9.5.0/includes/wc-template-functions.php on line 1668
``` 

That line got altered here in 9.5
https://github.com/woocommerce/woocommerce/pull/51936

### Expected behavior

No fatal error

### Actual behavior

In 9.5.0 if a shopper visits a variable product page that has used that plugin to add extra product images it triggers the fatal.
The page displays the product images instead (as if the plugin was deactivated)
But otherwise functionality seems to still work (just no additional images).

9.4.3 WC only
Product images and variation details
Image

Image

9.4.3 WC & WooCommerce Additional Variation Images plugin
I’ve added a green tick here as an extra image, and you can see it when you select that variation as a shopper
Image

Image

9.5.0 WC & WooCommerce Additional Variation Images plugin
Shopper doesn’t see extra image, functionality still works, Fatal appears in site logs.

Image

### Steps to reproduce

see actual behaviour above

### WordPress Environment

N/A

### Isolating the problem

- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #53761.

This PR adds a check to ensure `$product` is a `WC_Product` instance before invoking `get_title`. This change is necessary because some extensions, like [WooCommerce Additional Variation Images](https://woocommerce.com/it/products/woocommerce-additional-variation-images/), generate a new HTML fragment via REST API, where `$product` may not be a `WC_Product` instance.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you're using a block theme with a default Single Product Template.
2. Install WooCommerce Additional Variation Images https://woocommerce.com/it/products/woocommerce-additional-variation-images/
3. Ensure that `WP_DEBUG` is set to true.
4. Edit a product that has some variations (like V-Neck T-Shirt).
5. Click on variations.
6. Edit a variation
7. Add an additional image:

image
8. Save.
9. Visit that product.
10. Select that variation.
11. Ensure that the additional image is visible.
12. Ensure that no PHP fatal error is present in the logs.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Product Gallery: fix PHP fatal error.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/includes/wc-template-functions.php [hunk 1]
```diff
@@ -1665,7 +1665,7 @@ function wc_get_gallery_image_html( $attachment_id, $main_image = false, $image_
 	$thumbnail_sizes   = wp_get_attachment_image_sizes( $attachment_id, $thumbnail_size );
 	$full_src          = wp_get_attachment_image_src( $attachment_id, $full_size );
 	$alt_text          = trim( wp_strip_all_tags( get_post_meta( $attachment_id, '_wp_attachment_image_alt', true ) ) );
-	$alt_text          = empty( $alt_text ) ? woocommerce_get_alt_from_product_title_and_position( $product->get_title(), $main_image, $image_index ) : $alt_text;
+	$alt_text          = empty( $alt_text ) && is_product( $product ) ? woocommerce_get_alt_from_product_title_and_position( $product->get_title(), $main_image, $image_index ) : $alt_text;
```
Comment (gigitux) [inline plugins/woocommerce/includes/wc-template-functions.php]: If `$product` is null, is_product doesn't trigger any error.

Comment (tjcafferkey) [reply plugins/woocommerce/includes/wc-template-functions.php]: `is_product()` doesn't accept an argument AFAIK https://developer.woocommerce.com/docs/conditional-tags-in-woocommerce/

Comment (gigitux) [reply plugins/woocommerce/includes/wc-template-functions.php]: Nice catch! 😅 Fixed with the right method f1d4baada8e3b4b0f4ff51d804dbb922252303ef

**Review by tjcafferkey — APPROVED** [review]
This fixes the issue described, I'm not seeing any errors getting logged. LGTM!

**Comment by coreymckrill** [comment]
^ FYI @naman03malhotra the debugging link above is broken, looks like it has an extra `woocommerce` slug in it.

**Comment by coreymckrill** [comment]
Turns out the cherry pick to trunk is not needed because the issue was already fixed there in #53167.

**Comment by naman03malhotra** [comment]
Thanks @coreymckrill, I've created a PR to address this: https://github.com/woocommerce/woocommerce/pull/53824

**Comment by sbbasile** [comment]
Following this update, the cart is no longer empty.

**Pull request merged**